torch_np/_detail/_reductions.py

```diff
@@ -4,13 +4,78 @@
 Anything here only deals with torch objects, e.g. "dtype" is a torch.dtype instance etc
 """
 
+import typing
+
 import torch
 
 from . import _dtypes_impl, _util
 
 NoValue = None
 
 
+import functools
+
+############# XXX
+### From _util.axis_expand_func
+
+
+def deco_axis_expand(func):
+    """Generically handle axis arguments in reductions."""
+
+    @functools.wraps(func)
+    def wrapped(tensor, axis, *args, **kwds):
+
+        if axis is not None:
+            if not isinstance(axis, (list, tuple)):
+                if not isinstance(axis, typing.SupportsIndex):
+                    raise TypeError(
+                        f"{type(axis)=}, but should be a list/tuple or support operator.index()"
+                    )
+                axis = (axis,)
+            axis = _util.normalize_axis_tuple(axis, tensor.ndim)
+
+        if axis == ():
+            newshape = _util.expand_shape(tensor.shape, axis=0)
+            tensor = tensor.reshape(newshape)
```

Comment:
I had to look up what `expand_shape` does here. But doesn't this simply do
```suggestion
            tensor = tensor.reshape((1,))
```

Reply:
Not really. 

```
In [14]: t = torch.ones(2, dtype=int)

In [15]: t.reshape((1,))
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [15], in <cell line: 1>()
----> 1 t.reshape((1,))

RuntimeError: shape '[1]' is invalid for input of size 2

In [16]: from torch_np._detail import _util

In [17]: _util.expand_shape(t.shape, axis=0)
Out[17]: [1, 2]
```

What's going on here is that `expand_shape` is numpy's `expand_dims`, basically `torch.unsqueeze`. Here we insert a length-one axis and run the reduction along it. This is to emulate this numpy's beauty:

```
In [19]: np.sum(t.numpy(), axis=())
Out[19]: array([1, 1])
```

which is different from 

```
In [24]: np.sum(t.numpy(), axis=None)
Out[24]: 2
```

Reply:
Right. Thank you for the comment!